py-version-control.Rmd

```diff
@@ -0,0 +1,428 @@
+folder. Now if we make any changes in that folder (or a sub-folder!), git 
+will pay attention and be able to track it for us. Let's see what happens 
+if we do this. 
+
+## Tracking Changes {#py-version-control-tracking-changes}
+
+As we add and modify files in our project folder, we want to record our changes. 
+This is done by creating "commits" which are like snapshots of the repository at 
+moments in time. By building up a history of many commits, each created for a 
+different phase of the files in the repository, you can see what changes have happened over time. 
+
+The process of tracking changes in this way has three stages: 
+
+1. Create new files or make changes to existing files in the project repository.
+2. Choose the files that you want to create a saved version of (called "staging" or "adding")
+3. "Commit" the changes from the previous step, with a descriptive commit message. 
+
+![commit-cycle](figures/py-version-control/tmp-commit-cycle.jpeg)
+
+
+To see this in action, create a file called `README.md` in your test repository. 
+You can technically do this through GitKraken, but it would make more sense to 
+create this file in a text editor like Spyder - it should just be a plain text 
+document, saved in the main folder. 
+
+```
+# Carbon Levels Project
+
+## Meeting time
+
+- December 14, 2019
+- 18:00 UTC
+
+## Collaborators
+
+- Gina
+- Mark
+- Jesus
+
+## My notes
+
+```
+
+After creating this file, switch back to GitKraken and 
+look at the right side of the GitKraken interface. You should see a few changes 
+in the right hand pane. If you click on "View Change", the right pane will change to show you the list of files that git knows has changed. In this case, it should 
+have the new `README.md` cycle. 
+
+It's now time to do the second part of the cycle -- *adding* or *staging* files to commit. In the GitKraken interface, this can be done by hitting the "stage all changes" button (which will add *all* the changed files to the list to be committed) or you can click on individual files and then click on the "Stage File" button. 
+
+Once you've done this, the files that are to be added to the commit will be now in the second box of the right pane, the staging area. If you made a mistake and added files you aren't yet ready to snapshot, you can "unstage" them here. Once you have the list of files you want to stage, write a commit message in the third box. Then click on the commit changes to one file. 
+
+We're sent back to our history mode, with the right pane showing us the latest commit. We've gone through the change-and-commit cycle one time. Git has saved 
+information about the state of our `README.md` file in the repository we created 
+in the previous section. Now, if we make additional changes to the `README.md` 
+file, we'll be able to create new commits, building up a history of changes. Try 
+making changes to `README.md` under the "My Notes" section 
```

Comment:
Maybe this challenge should have different formatting to the regular text? (Otherwise the text just kind of flows into the challenge and it's a little confusing.) For now it could just be an indented code block? e.g.

> **Repeat the cycle**
>
> Try making changes to `README.md`...

Reply:
It was intended to be a "challenge" per se, but I'll split it into its own paragraph for sure.